frontend/src/routes/courses/+page.svelte

```diff
@@ -0,0 +1,93 @@
+<script lang="ts">
+  import { onMount } from 'svelte';
+  import { writable } from 'svelte/store';
+  import { get } from '$utils/fetch';
+  import { _ } from 'svelte-i18n';
+
+  interface Course {
+    name: string;
+    user: {
+      first_name: string;
+    };
+  }
+
+  let courses = writable<Course[]>([]);
+  let search = '';
+
+  async function submitForm() {
+    const query = encodeURIComponent(search.trim());
+    window.location.href = `/courses?search=${query}`;
```

Comment:
It's better to use [goto](https://kit.svelte.dev/docs/modules#$app-navigation-goto) here to prevent full page reloading.